solution_7

A-Glebov opened the pull request:
1. Добавили класс FileBackedTaskManager для сохранения текущего состjяния менеджера в файл

2. Добавили Enum для типов задач
3. Добавили тип задачи в конструктор
4. Добавили тестирование записи и чтения пустого файла и сценарий для нескольких задач разных типов

A-Glebov committed:
1. Добавили класс FileBackedTaskManager для сохранения текущего состояния менеджера в файл
2. Добавили Enum для типов задач
3. Добавили тип задачи в конструктор
4. Добавили тестирование записи и чтения пустого файла и сценарий для нескольких задач разных типов

BerezinYaroslav commented on src/service/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,192 @@
+import java.nio.file.Files;
+import java.util.ArrayList;
+
+public class FileBackedTaskManager extends InMemoryTaskManager {
```
⛔️ В Managers в методе getDefault нужно поменять возвращаемую реализацию с InMemory на эту

BerezinYaroslav replied:
✅

BerezinYaroslav commented on src/service/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,192 @@
+
+    }
+
+    static FileBackedTaskManager loadFromFile(File file) {
```
⛔️ Нужно добавить модификатор доступа этому и другим методам, где его нет

BerezinYaroslav replied:
✅

BerezinYaroslav commented on src/service/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,192 @@
+        this.file = file;
+    }
+
+    public static void main(String[] args) throws IOException {
```
⛔️ Знаю, что в ТЗ сказано сделать этот метод именно в этом классе, но лучше перенести его в класс Main, так как именно там происходит запуск программы и "прогонка" по тестовым сценариям. Размещать подобные методы в сервисах, как этот, - плохая практика. В ТЗ это сделано для упрощения (что, в целом, на данном этапе ок)

BerezinYaroslav replied:
✅

BerezinYaroslav commented on src/service/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,192 @@
+    }
+
+    static FileBackedTaskManager loadFromFile(File file) {
+        FileBackedTaskManager fileBackedTaskManager = new FileBackedTaskManager(file);
```
⛔️ Тебе не нужен здесь экземпляр этого же класса. Если тебе нужно к нему обращаться, используй this

BerezinYaroslav replied:
Обсудили, пусть остается

BerezinYaroslav commented on src/service/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,192 @@
+                fileWriter.write(toString(task) + '\n');
+            }
+
+        } catch (IOException e) {
```
✅ Есть обработка

BerezinYaroslav commented on test/service/FileBackedTaskManagerTest.java:
```diff
@@ -0,0 +1,57 @@
+
+class FileBackedTaskManagerTest {
+
+    File file;
```
⛔️ У всех-всех полей и методов всех-всех классов должны быть модификаторы доступа. По всем классам пройдись и поправь

Кстати, при объявлении класса (в этом классе это строка 13) тоже нужен модификатор доступа, это всегда public, если только это не вложенный класс

BerezinYaroslav replied:
✅

BerezinYaroslav reviewed (changes requested):
Алексей, привет!

Хорошая работа, но есть важные для исправления моменты. Если будут какие-то вопросы - пиши мне в Пачке

Удачи!

A-Glebov committed:
1. Добавили модификаторы доступа
2. В Managers создали метод для создания FileBackedTaskManager
3. Вынесли пользовательский сценарий из FileBackedTaskManager в Main

BerezinYaroslav reviewed (approved):
Алексей, привет! Все хорошо, принято!

Удачи в новом спринте)
